### PR TITLE
feat(ui, api): add support for non image message attachments, make s3 attachments work

### PR DESCRIPTION
**Pull request by JoltCode**
# Description

Adds support for downloading non image file attachments - also returns files in message result page api calls

Fixes N/A (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

N/A - Altho @RhysSullivan please can we check over the perms for attachments - if the user hasn't consented 👍

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have updated .env.example if I added a new environment variable
- [x] My PR title follows the [semantic commits](https://www.conventionalcommits.org/en/v1.0.0/) style

### CODE REVIEW

File: packages/env/shared.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ export const sharedEnvs = createEnv({
 		ELASTICSEARCH_PASSWORD: z.string(),
 		ELASTICSEARCH_USERNAME: z.string(),
 		ELASTICSEARCH_MESSAGE_INDEX: z.string(),
+		ELASTIC_DISABLED: z.string().optional(),
```
Comment (RhysSullivan) [inline packages/env/shared.ts:92]: Could you remove this and I'll fix it in a different PR

File: packages/db/src/pages.ts [hunk 2]
```diff
@@ -313,6 +314,12 @@ export async function findMessageResultPage(
 		}),
 	]);
 
+	const allMessageIds = messages.map((m) => m.id);
+
+	const attachments = await db.query.dbAttachments.findMany({
```
Comment (RhysSullivan) [inline packages/db/src/pages.ts:319]: Let's keep attachments on the message, not separate - if it's a bug let's fix the bug

File: packages/ui/src/components/primitives/message/AttachmentDownloader.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export const AttachmentDownloader = (props: AttachmentDownloaderProps) => {
+	const fileNameTruncated =
+		props.filename.length > 20
+			? props.filename.substring(0, 20) + '...'
+			: props.filename;
+
+	return (
+		<LinkButton
+			className="flex flex-row items-center justify-start"
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/AttachmentDownloader.tsx:19]: Let's make sure light mode + mobile

File: packages/ui/src/components/primitives/message/attachments.tsx [hunk 4]
```diff
@@ -41,26 +42,53 @@ const SingularImageAttachment = async (
...
 	message.attachments = message.attachments.filter((attachment) =>
 		imageFileRegex.test(attachment.filename.toLowerCase()),
 	);
 	if (message.attachments.length === 0) return null;
-	if (props.limit)
-		message.attachments = message.attachments.slice(0, props.limit);
 
-	// TODO: Rename this and such, whole file needs a revisit but I'm on vacation 🌴
 	return (
 		<SingularImageAttachment
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/attachments.tsx:57]: Can we rename this in this PR

File: packages/ui/src/components/primitives/message/attachments.tsx [hunk 5]
```diff
@@ -41,26 +42,53 @@ const SingularImageAttachment = async (
...
+
+	if (message.attachments.length === 0) return null;
+	if (props.limit)
+		message.attachments = message.attachments.slice(0, props.limit);
+	const imageFileRegex = new RegExp('(.*/)*.+.(png|jpg|gif|bmp|jpeg|webp)$');
+	const imageAttachments = message.attachments.filter((attachment) =>
+		imageFileRegex.test(attachment.filename.toLowerCase()),
+	);
+	if (imageAttachments.length > 0) {
+		return <ImageAttachments message={message} />;
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/attachments.tsx:79]: Make sure multi attachment types work here, image + files should
Show both

File: packages/ui/src/components/primitives/message/attachments.tsx [hunk 6]
```diff
@@ -41,26 +42,53 @@ const SingularImageAttachment = async (
...
+	}
+
+	// TODO: Rename this and such, whole file needs a revisit but I'm on vacation 🌴
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/attachments.tsx:82]: Can remove comment

File: packages/ui/src/components/primitives/message/attachments.tsx [hunk 7]
```diff
@@ -41,26 +42,53 @@ const SingularImageAttachment = async (
...
+	return (
+		<div className="grid grid-cols-1 gap-4 pt-4 md:grid-cols-2 xl:grid-cols-3 2xl:grid-cols-4">
```
Comment (RhysSullivan) [inline packages/ui/src/components/primitives/message/attachments.tsx:84]: Make sure mobile + light mode

**Review by RhysSullivan — CHANGES_REQUESTED** [review]

File: packages/db/src/pages.ts (hunk 2 above)
Comment (JoltCode) [reply packages/db/src/pages.ts]: Fixed the bug! ^ 😄

File: packages/ui/src/components/primitives/message/attachments.tsx (hunk 7 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/attachments.tsx:84]: Yep! Works perfect in light mode + mobile! 😄

File: packages/ui/src/components/primitives/message/AttachmentDownloader.tsx (hunk 3 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/AttachmentDownloader.tsx:19]: Yep, works well in both! 👍

File: packages/ui/src/components/primitives/message/attachments.tsx (hunk 4 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/attachments.tsx:57]: 👍

File: packages/ui/src/components/primitives/message/attachments.tsx (hunk 6 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/attachments.tsx:82]: 👍

File: packages/ui/src/components/primitives/message/attachments.tsx (hunk 5 above)
Comment (JoltCode) [reply packages/ui/src/components/primitives/message/attachments.tsx]: Fixed! 😄🚀

File: packages/env/shared.ts (hunk 1 above)
Comment (JoltCode) [reply packages/env/shared.ts]: Done! 👍

**Pull request merged**